### ISSUE TITLE
[Question] Can this library be used in .net framework 4.7.2 with basic authentication?

### ISSUE THREAD

**Issue by victoriorinescu** [protobuf-net/protobuf-net.Grpc#101]
I am trying to consume the grpc service from .net framework 4.7.2.
However, I don't know how to set the HttpClient.

The following code works perfect in .net core:
`            GrpcClientFactory.AllowUnencryptedHttp2 = true;
            var handler = new HttpClientHandler();
            var httpClient = new HttpClient(handler);
            handler.Credentials = new NetworkCredential("admin", "pass", "");
            var channel = GrpcChannel.ForAddress("http://localhost:10000", new GrpcChannelOptions {
                HttpClient = httpClient,            })`

However, for .net framework nothing seems to work:

`            GrpcClientFactory.AllowUnencryptedHttp2 = true;
            var channel = new Channel("localhost", 10000, ChannelCredentials.Insecure);

            var credentials = CallCredentials.FromInterceptor((context, metadata) =>
            {
                metadata.Add("Authorization", "Basic QWxhZGRpbjpPcGVuU2VzYW1l");
                return Task.CompletedTask;
            });

            CallOptions co = new CallOptions().WithCredentials(credentials);
            Metadata md = new Metadata();
            md.Add("Authorization", "Basic QWxhZGRpbjpPcGVuU2VzYW1l");
            co.WithHeaders(md);

            var service = channel.CreateGrpcService<ICommonServiceGrpc>();
            var people = await service.GetPeople(co);`

Any idea?

**Comment by mgravell** [comment]
does that need to be `co = co.WithHeaders(md);` ?

**Comment by victoriorinescu** [comment]
not necessarily; it is extra because credentials does not seem to work - so I just tried with headers; does not work also.
